src/web100srv.c

```diff
@@ -1466,74 +1352,120 @@ void child_process(int parent_pipe, SSL_CTX *ssl_context, int ctlsockfd) {
   ctl.socket = ctlsockfd;
   // this is the child process, it handles the connection with the client and
   // runs the actual tests.
-  log_println(4, "Child thinks pipe() returned fd0=%d, fd1=%d for pid=%d",
-              chld_pipe[0], chld_pipe[1], chld_pid);
-  close(listenfd);
-  close(chld_pipe[1]);
+  // First, start the watchdog timer
+  alarm(300);
+
+  // Set up the connection with the client (with optional SSL and websockets)
+  if (options.tls) {
+    ctl.ssl = SSL_new(ssl_context);
+    if (ctl.ssl == NULL) {
+      log_println(4, "SSL_new failed");
+      exit(-1);
+    }
+    if (SSL_set_fd(ctl.ssl, ctl.socket) == 0) {
+      log_println(4, "SSL_set_fd failed");
+      exit(-1);
+    }
+    if ((ssl_err = SSL_accept(ctl.ssl)) != 1) {
+      if (ssl_err == 0) {
+        ssl_err = SSL_get_error(ctl.ssl, ssl_err);
+      }
+      ERR_error_string_n(ssl_err, ssl_error, sizeof(ssl_error));
+      log_println(4, "SSL_accept failed (%s)", ssl_error);
+      exit(-1);
+    }
+  }
```

Comment:
Can we refactor this to remove the redundancy between this and `setup_SSL_connection`?